File: valid_mountain_array.js
const valid_mountain_array = array => {
    const lens = array.length
    let i = 0
    while (i < lens && array[i] < array[i+1]) i++
    if(i === 0 || i === lens -1) return false
    while (i < lens && array[i] > array[i+1]) i ++
    return i === lens - 1
}


let b = [12, 23, 34, 45, 34, 23, 67]

console.log(valid_mountain_array(b))

File: valid_mountain_array.rs
fn valid_mountain_array(a: Vec<i32>) -> bool {
    let lens = a.len();
    let mut i = 0;
    if lens < 3 {
        return false;
    }
    while i < lens - 1 && a[i] < a[i+1] {
        i += 1
    }
    if i == 0 || i == lens - 1 {
        return false;
    }

    while i < lens - 1 && a[i] > a[i+1] {
        i += 1
    }

    return i == lens - 1;
}